Expression filters

tobrun opened the pull request:
WIP, this PR provides Android binding integration for #11251: 
 - [x] remove old Filter.java API
 - [x] integrate `setFilter(Expression filter)`
    - [x] update `MapboxMap#queryRenderedFeatures`
    - [x] update `Source#querySourceFeatures` on source types
    - [x] update `Layer#setFilter` on layer types (code-gen) 
 - [x] add `getFilter` 
   - [x] manage state corretly when no filter is availlable (java.lang.Error: in get<T>())
 - [x] integrate generated instrumentation test for set/get filter
 - [x] migrate test app code to expression syntax

ivovandongen commented on platform/android/src/style/layers/layer.hpp:
```diff
@@ -68,6 +68,8 @@ class Layer : private mbgl::util::noncopyable {
 
     void setFilter(jni::JNIEnv&, jni::Array<jni::Object<>>);
 
+    jni::Object<jni::ObjectTag> getFilter(jni::JNIEnv&);
```
Can we type this more specific than `Object`?

tobrun commented:
@ivovandongen thank you for the review, I addressed your comments in https://github.com/mapbox/mapbox-gl-native/pull/11429/commits/145e1e09d64cf1d1e1533cfc892e12aabfedddec. (FWIW flagging that most of the code in this PR will be refactored out with https://github.com/mapbox/mapbox-gl-native/issues/11419 and https://github.com/mapbox/mapbox-gl-native/issues/11420)

vanshg commented:
I see that Filter.none has been removed with no explicit replacement (as well as Filter.notHas). Are there any plans on making those explicitly available, or will we have to with something more verbose such as Expression.not(Expression.any) and Expression.not(Expression.has)?

tobrun commented:
> I see that Filter.none has been removed with no explicit replacement (as well as Filter.notHas). Are there any plans on making those explicitly available, or will we have to go with something more verbose such as Expression.not(Expression.any) and Expression.not(Expression.has)?

the latter
